lib/internal/fs/dir.js

```diff
@@ -94,7 +96,7 @@ class Dir {
   }
 
   read(callback) {
-    return this.#readImpl(true, callback);
+    return arguments.length === 0 ? this.#readPromisified() : this.#readImpl(true, callback);
```

Comment:
I've moved this change to https://github.com/nodejs/node/pull/58228, let's wait for that to land first